### ISSUE TITLE
Not an issue, just learning and curious

### ISSUE THREAD

**Issue by JHansen13** [SlavicPotato/ied-dev#28]
I've been reviewing the source code to learn more about SKSE plugin development.

I'm wondering where I could locate the source code for ITaskPool? It looks like it's a part of the precompiled header, but that's the only place I've been able to locate it. Is it from an external library of some source?

**Comment by SlavicPotato** [comment]
> Is it from an external library of some source

Yeah, it's part of a library used for my plugins. I recently switched most stuff to MIT license, however that repo also contains some code from skse resources which isn't compatible with the license so I've privated it for now. It's not a general purpose library though, it's tailored to whatever I might need, often the changes are breaking so I'd discourage anyone from using it in any case (see commonlib instead)

That said, ITaskPool is just a simple task scheduler similar to what skse adds to allow scheduling tasks where BSTaskPool runs, the difference is that the hook is installed on the main game loop and it supports both one time and repeating (per-frame) tasks.

**Comment by JHansen13** [comment]
Thank you. That's a great explanation. Do you have a discord or other location where you would be willing to answer more questions on SKSE dev?
